File: GolangAns/Easy/reverse_Intenger.go
package main

func main() {
	println(reverse(-123))
}

func reverse(x int) int {
	var pop int
	reversed := 0
	var MaxInt int32 = 2147483647

	if x > int(MaxInt) || x < -int(MaxInt) {
		return 0
	}
	for x != 0 {
		pop = x % 10
		x = x / 10

		reversed = (reversed * 10) + pop
	}

	if reversed > int(MaxInt) {
		return 0
	}
	return reversed

}

File: PythonAns/reverseIntenger.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == 0:
            return 0

        ret = 1
        if x < 0:
            ret *= -1

        retStr = ''
        str_ = str(x)
        mark = False
        for i in reversed(str_):
            if (i == '0' and mark is False) or i == '-':
                continue
            retStr += i
            mark = True

        ret *= int(retStr)

        if ret < (-2)**31 or ret > (2**31)-1:
            return 0

        return ret
